vnext/ReactUWP/Modules/AsyncStorageModuleWin32.cpp

```diff
@@ -0,0 +1,191 @@
+#include "pch.h"
+
+#include "AsyncStorageModuleWin32.h"
+
+#include <cstdio>
+
+/// Implements ASM using winsqlite3.dll (requires Windows version 10.0.10586)
+
+namespace facebook {
+namespace react {
+
+namespace {
+
+static void ExecImpl(sqlite3 *db, const char *statement, AsyncStorageModuleWin32::ExecCallback callback, void *pv) {
+  char *errMsg = nullptr;
+  int rc = sqlite3_exec(db, statement, callback, pv, &errMsg);
+  if (errMsg) {
+    throw std::runtime_error(errMsg);
+  }
+  if (rc != SQLITE_OK) {
+    throw std::runtime_error(sqlite3_errmsg(db));
+  }
+}
+
+class Sqlite3Transaction {
+  sqlite3 *m_db{nullptr};
+
+ public:
+  explicit Sqlite3Transaction(sqlite3 *db) : m_db(db) {
+    if (db) {
+      ExecImpl(db, u8"BEGIN TRANSACTION", nullptr, nullptr);
+    }
+  }
+  Sqlite3Transaction(const Sqlite3Transaction &) = delete;
+  Sqlite3Transaction(Sqlite3Transaction &&other) : m_db{other.m_db} {
+    other.m_db = nullptr;
+  }
+  Sqlite3Transaction &operator=(const Sqlite3Transaction &) = delete;
+  Sqlite3Transaction &operator=(Sqlite3Transaction &&rhs) {
+    std::swap(m_db, rhs.m_db);
+  }
+  ~Sqlite3Transaction() {
+    ExecImpl(m_db, u8"COMMIT", nullptr, nullptr);
+  }
+};
+
+} // namespace
+
+AsyncStorageModuleWin32::AsyncStorageModuleWin32(PCSTR dbPath) {
+  checkSQLiteResult(
+      sqlite3_open_v2(dbPath, &m_db, SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE | SQLITE_OPEN_FULLMUTEX, nullptr));
+
+  int userVersion = 0;
+  auto userVersionCallback = [&](int cCol, char **rgszColText, char **rgszColName) {
+    userVersion = atoi(rgszColText[0]);
+    return SQLITE_OK;
+  };
+
+  Exec(u8"PRAGMA user_version", userVersionCallback);
+
+  if (userVersion == 0) {
+    Exec(u8"CREATE TABLE IF NOT EXISTS AsyncLocalStorage(key TEXT, value TEXT); PRAGMA user_version=1");
+  }
+}
+
+AsyncStorageModuleWin32::~AsyncStorageModuleWin32() {
+  checkSQLiteResult(sqlite3_close(m_db));
+}
+
+std::string AsyncStorageModuleWin32::getName() {
+  return "AsyncLocalStorage";
```

Comment:
FYI this is for the deprecated built in version of Asyncstorage. If we want compatibility with the post-lean-core we'll need to change this to PlatformLocalStorage.

Reply:
I'm not familiar with `PlatformLocalStorage`, what's that? I expected that we would move this out to [react-native-community](https://github.com/react-native-community/async-storage) at some point, yes.

Reply:
They changed the expected name of the Native module. The interface is still the same, but the new JS will search for `PlatformLocalStorage` instead of `AsyncLocalStorage`.

Reply:
OK, this sounds like something we'll do when we move this module out to the community, right/

Reply:
Yep, that sounds right